src/actions/actionClipboard.tsx

```diff
@@ -131,10 +130,19 @@ export const actionCopyAsPng = register({
 export const copyAllTextNodesAsText = register({
   name: "copyAllTextNodesAsText",
   trackEvent: { category: "element" },
-  perform: (elements) => {
-    const text = (
-      getNonDeletedElements(elements) as ExcalidrawTextElement[]
-    ).reduce((acc, element) => `${acc}${element.text}\n`, "");
+  perform: (elements, appState) => {
+    const selectedElements = getSelectedElements(
+      getNonDeletedElements(elements),
+      appState,
+      true,
+    );
+
+    const text = selectedElements.reduce((acc, element) => {
+      if (element.type === "text") {
```

Comment:
```suggestion
      if (isTextElement(element)) {
```
since we have a util already